exp-models/addon/utils/eligibility.js

```diff
@@ -0,0 +1,113 @@
+// h/t @aaxelb
+
+import Ember from 'ember';
+import moment from 'moment';
+
+let {$} = Ember;
+
+var AND = (a, b) => {
+    return function() {
+        return a(...arguments) && b(...arguments);
+    };
+};
+var OR = (a, b) => {
+    return function() {
+        return a(...arguments) || b(...arguments);
+    };
+};
+
+/**
+ * @return {integer}: number of days represented by amount/unit pair, e.g. 3 'weeks' -> 21
+ **/
+function parseDate(amount, unit='days') {
+    var singular = unit.replace(/s$/, '');
```

Comment:
I would sugggest using `Ember.Inflector.inflector.singularize(word)`, or just skipping this line and perhaps using `pluralize` directly. (you seem to be making something singular in order to pluralize)
http://emberjs.com/api/data/classes/Ember.Inflector.html

Ember's inflector is designed to handle more examples of irregular plurals, and using it would let us reuse any words that had already been registered as `uncountable` etc. 

Basically, just a matter of standardizing on existing logic instead of rewriting our own. :)